go-controller/pkg/config/config.go

```diff
@@ -455,7 +455,7 @@ var CommonFlags = []cli.Flag{
 	// Logging options
 	cli.IntFlag{
 		Name:        "loglevel",
-		Usage:       "log verbosity and level: 5=debug, 4=info, 3=warn, 2=error, 1=fatal (default: 4)",
+		Usage:       "klog log verbosity and level [0-10]: > 5 extreme verbosity, 5=debug, < 5 info, warning, error (default: 4)",
```

Comment:
@alexanderConstantinescu the "< 5 info, warning, error" bit is still really confusing. It's not even technically true :) As it stands this PR only uses levels 5 (debug) and 1 (info, warning, error).

But in the end klog adds its own flags regardless of what CLI flags we have, and you can set those without touching our --loglevel. So what if to preserve backwards compatibility, we do this:

`Usage:       "[deprecated, use klog flags instead] log verbosity and level: 5=debug, 4=info, 3=warn, 2=error, 1=fatal (default: 4)"`

eg keep the simple existing loglevel for a while, and then map that to klog:

```
switch Logging.Level {
case 1:
    err = klog.V(1).Set("fatal")
case 2:
    err = klog.V(1).Set("error")
case 3:
    err = klog.V(1).Set("warning")
case 4:
    err = klog.V(1).Set("info")
case 5:
    var level klog.Level
    err = level.Set(5)    
}
<do something intelligent with 'err'>
```

Then eventually we remove our custom --loglevel.

Reply:
We cannot do that due to what I mentioned in the description to this PR: the cli package `github.com/urfave/cli` implements the flag `-v =  version`, and there's no way to override that flag definition for `klog` (which expects `-v = verbosity`)

Our `loglevel` flag thus sits as a wrapper around `-v`. 

I will update the explanation with this and what you mentioned.

Reply:
@alexanderConstantinescu the CLI help text is still really confusing. Yes, it's technically correct. No, it doesn't answer the question "I want debug logging, what level do I choose?" without further reference. The help text should provide a clear answer to that question.

We can still do what I pasted above for mapping our log level intelligently to klog, but let's just skip the [deprecated] thing in the help text then. eg

`Usage: "log verbosity and level: 5=debug, 4=info, 3=warn, 2=error, 1=fatal (default: 4)"`

and then the mapping. And as you state we just use our loglevel and don't use klog because the urfave thing.

Reply:
Concepts are being mixed here. You have the concept of severity (INFO, WARN, ERR, etc) and log level. All severity is _always_ printed, no matter what log-level is specified and you have no way around that (i.e `klog.V(1).Set("fatal")` is not possible - that method `V(something).Set` does not exist)

The only logging control klog provides is: `klog.V(X).Infof(msg)` which prints the log line if klog's log level equals `X`. In this case (i.e with this patch) we define that debug = 5, but this is a "user defined log level", we cannot do that for info, warning, error, fatal as those are klog "built ins". I chose debug = 5 as to allow anyone using ovn-kubernetes (CNO for ex) to be backwards and forwards compatible w.r.t this patch and using the log level debug. 

I can thus only propose to change the description again, how about: 

`Usage: "log verbosity and level:  info, warn, fatal, error are always printed no matter the log level. Use 5 for debug (default: 4)."`

Reply:
@alexanderConstantinescu I'm arguing that having separate severity and level is fundamentally confusing and we shouldn't do that exactly. For example, I don't see any reason why we should accommodate `-v=5 --level=warning`. That makes no sense at all, but it's possible with klog.

Let's just keep it simple and have levels, and emulate them with klog.

Reply:
@dcbw I don't know why anyone would filter out warning, error, or fatal messages. They are very critical for any deployers to ascertain the health of these daemons

```
Usage: "log verbosity and level: 5=debug, 4=info, 3=warn, 2=error, 1=fatal (default: 4)"
```
With `klog`, as I see it, there are only two broader log types -- Info and rest (error, warn, and fatal). For, Info we can have several levels. To keep backwards compatibility:
```
logrus.Info() --> klog.V(4).Info
logrus.Debug() --> klog.V(5).Info
logrus.Error() --> klog.Error()
logrus.Warn() --> klog.Warning()
logrus.Fatal() --> klog.Fatal()
```
So, with this we will support only --level=5, --level=4, and --level={3,2,1} will all have same effect and will always log warn, error, and fatal.

Reply:
`--loglevel` should work exactly the way it does in other klog-using modules. It's confusing to have a component in the kubernetes ecosystem that has a `--loglevel` argument that does not behave the way that it does in other components of the kubernetes ecosystem.

That means the behavior of `--loglevel X` is that it makes `klog.V(Y).Info()` log for Y>=X and not log for Y<X. If we are not using some values of Y in the code then that means that some values of X are not useful as command-line arguments but this is true for all modules that use klog.

Fatals, Errors, and Warnings should always be logged.

Reply:
> That means the behavior of --loglevel X is that it makes klog.V(Y).Info() log for Y>=X and not log for Y<X.

It's the other way around, right? If X = 4 and Y = 5, then those `klog.V(5).Info()` logs _won't_ be printed. 

Besides that I agree with both you and @girishmg (though I don't see the point in doing `logrus.Info() -> klog.V(4).Info`. `logrus.Info() -> klog.Info()` will have the same effect and is more understandable for future developers of ovn-kubernetes).

Reply:
> It's the other way around, right? If X = 4 and Y = 5, then those `klog.V(5).Info()` logs _won't_ be printed.

Yes, I said it backwards.



> I don't see the point in doing `logrus.Info() -> klog.V(4).Info`. `logrus.Info() -> klog.Info()` will have the same effect - if we set the default loglevel to 4

Yeah, I think `logrus.Info()` should just become `klog.Info()`. It should absolutely not become `klog.V(4).Info`. 4 is closer to debug than to info.

I believe the expectation with klog is that production clusters will normally run with log level 2. Basically:

- `klog.Info` / `klog.V(1).Info` - informational message you always care about
- `klog.V(2).Info` - informational message you usually care about
- `klog.V(3).Info` - :man_shrugging: 
- `klog.V(4).Info` - messages you only care about when debugging, which won't fill up your logs too quickly
- `klog.V(5).Info` - spam

Reply:
So, this patch _almost_ follows that convention. We're only missing the diff between "logs which won't fill up too quickly" and "spam". 

As I mentioned though, I find it difficult to decide which logs should be changed from `klog.Info` to `klog.V(X).Info`, as I didn't write the code for the majority of those logs. I would thus stick to having this patch simply convert `logrus.Info` to `klog.Info`.

Reply:
Yes, it's fine for now to convert all `logrus.Info` to `klog.Info` and all `logrus.Debug` to `klog.V(4).Info` (or 5 if you want). We can adjust things later.